dotCMS/src/main/java/com/dotcms/util/AssetExporterUtil.java

```diff
@@ -0,0 +1,185 @@
+        final String hostName = Try
+                .of(() -> APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false).getHostname())
+                .getOrElse("dotcms");
+        final String fileName = StringUtils.sanitizeFileName(hostName)
+                + "_assets_"
+                + DateUtil.EXPORTING_DATE_FORMAT.format(new Date())
+                + ".tar.gz";
+        Logger.info(AssetExporterUtil.class, "User : " + user.getEmailAddress() + " downloading assets");
+
+        synchronized (AssetExporterUtil.class) {
+            try {
+                exportAssets(outputStream);
+            } catch (IOException e) {
+                e.printStackTrace();
+                throw new DotRuntimeException(e);
+            }
+        }
+
+        return fileName;
+    }
+    
+    /**
+     * Builds the tar.gz file containing the assets.
+     * 
+     * @param out output stream
+     */
+     static void exportAssets(OutputStream out) throws IOException {
+        try (TarArchiveOutputStream taos =
+                     new TarArchiveOutputStream(new GZIPOutputStream(new BufferedOutputStream(out)))) {
+            taos.setBigNumberMode(TarArchiveOutputStream.BIGNUMBER_STAR);
+            taos.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU);
+
+            FileUtil.listFilesRecursively(PARENT.get(), FILE_FILTER)
+                    .stream()
+                    .filter(File::isFile)
+                    .forEach(file -> {
+                        final String relativeFilePath = file.getPath().replace(PARENT.get().getPath(), "assets");
+                        TarArchiveEntry tarEntry = new TarArchiveEntry(file, relativeFilePath);
+                        tarEntry.setSize(file.length());
+                        try {
+                            taos.putArchiveEntry(tarEntry);
+                            try (InputStream in = Files.newInputStream(file.toPath())) {
+                                taos.write(IOUtils.toByteArray(in));
+                            }
+                            taos.closeArchiveEntry();
+                        } catch (Exception e) {
+                            throw new DotRuntimeException(e);
+                        }
+                    });
+        }
+    }
+
+    // TODO: verify if this is still the case
+    /**
+     * Unused at this time, intended to load a list of live/working inodes to check against so old
+     * versions do not get exported.
+     * 
+     * @param version version
+     * @param lastCursorId last cursor id
+     * @return loaded inodes
+     */
+    static Tuple2<String, Set<String>> loadUpInodes(final EXPORT_VERSION version, final String lastCursorId) {
+        final DotConnect db = new DotConnect().setSQL(LIVE_WORKING_SQL).addParam(lastCursorId);
+        final Set<String> inodes = new HashSet<>();
+        final StringWriter nextCursor = new StringWriter();
+        try {
+            db.loadObjectResults()
+                    .forEach(m -> {
```

Comment:
rename m to something else such as content version

Reply:
Renamed